### PR TITLE
Fix issue with expecting default ignore file when checking file size 

### PR DESCRIPTION
**Pull request by lnenad**
This PR fixes issue with expecting default ignore file when checking file size and adds case to check custom ignore file size. The issue was made by my previous PR https://github.com/jeremyjh/dialyxir/pull/543 due to the fact it was always trying to `File.stat!` the default ignore file. 

I've manually tested the following 5 combinations: 

* no ignore file at all no config
* default ignore file no config not empty
* default ignore file no config empty
* non-default ignore file path in config not empty
* non-default ignore file path in config empty

They all work but please @jeremyjh take a look. I'd also be happy to write tests for this but for some reason tests are failing even if I go back in history before my previous PR, can you please take a look if it is failing for you? Maybe it's something with my setup.

### CODE REVIEW

**Comment by jeremyjh** [comment]
@lnenad the tests do all pass for me locally. Can you paste a failure message? Yes I'd prefer to have tests and I think it should be possible following our project fixture patterns.

**Comment by jeremyjh** [comment]
@lnenad  The issue with some of those is the line terminators are different on windows; its true the test suite isn't compatible with windows.  I'm not sure what the story is with format vs format_warning though; if that is a platform difference it is a surprising one. Any chance you can test in WSL or a VM with Linux?

**Comment by jeremyjh** [comment]
@lnenad I think actually that those issues shouldn't prevent you from making a new test. If you can make a new test that passes for you in windows and fails with the old code, I can handle any issues with Linux compatibility/CI.

File: test/fixtures/ignore_empty/mix.exs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+defmodule Ignore.Mixfile do
```
Comment (riverrun) [inline test/fixtures/ignore_empty/mix.exs:1]: Not sure, but there might be an issue with using the same module name `Ignore.Mixfile`. Maybe you can try changing one of these names, and see if that fixes some of these warnings.

**Comment by pdm-jd** [comment]
I was linked to this PR after posting mine here https://github.com/jeremyjh/dialyxir/pull/549

It looks like a similar approach, but I'm not sure how to fix the failures in this one.

**Pull request closed without merging**